lib/autoupdate.ps1

```diff
@@ -351,69 +379,20 @@ function get_version_substitutions([String] $version, [Hashtable] $customMatches
 
 function autoupdate([String] $app, $dir, $json, [String] $version, [Hashtable] $matches) {
     Write-Host -f DarkCyan "Autoupdating $app"
-    $has_changes = $false
-    $has_errors = $false
-    [Bool]$valid = $true
     $substitutions = get_version_substitutions $version $matches
 
-    if ($json.url) {
-        # create new url
-        $url   = substitute $json.autoupdate.url $substitutions
-        $valid = $true
-
-        if($valid) {
-            # create hash
-            $hash = get_hash_for_app $app $json.autoupdate.hash $version $url $substitutions
-            if ($null -eq $hash) {
-                $valid = $false
-                Write-Host -f DarkRed "Could not find hash!"
-            }
-        }
-
-        # write changes to the json object
-        if ($valid) {
-            $has_changes = $true
-            update_manifest_with_new_version $json $version $url $hash
-        } else {
-            $has_errors = $true
-            throw "Could not update $app"
-        }
-    } else {
-        $json.architecture | Get-Member -MemberType NoteProperty | ForEach-Object {
-            $valid = $true
-            $architecture = $_.Name
-
-            # create new url
-            $url   = substitute (arch_specific "url" $json.autoupdate $architecture) $substitutions
-            $valid = $true
-
-            if($valid) {
-                # create hash
-                $hash = get_hash_for_app $app (arch_specific "hash" $json.autoupdate $architecture) $version $url $substitutions
-                if ($null -eq $hash) {
-                    $valid = $false
-                    Write-Host -f DarkRed "Could not find hash!"
-                }
-            }
-
-            # write changes to the json object
-            if ($valid) {
-                $has_changes = $true
-                update_manifest_with_new_version $json $version $url $hash $architecture
-            } else {
-                $has_errors = $true
-                throw "Could not update $app $architecture"
-            }
-        }
-    }
-
     # update properties
-    update_manifest_prop "extract_dir" $json $substitutions
-
-    # update license
-    update_manifest_prop "license" $json $substitutions
+    $properties_updated = @(coalesce ($json.autoupdate.PSObject.Properties.Name -ne 'architecture') @())
```

Comment:
Doesn't work when there is only one autoupdate property.

Kitty example:
```json
    "autoupdate": {
        "url": [
            "https://github.com/cyd01/KiTTY/releases/download/v$version/genpass.exe",
            "https://github.com/cyd01/KiTTY/releases/download/v$version/kageant.exe",
            "https://github.com/cyd01/KiTTY/releases/download/v$version/kitty.exe",
            "https://github.com/cyd01/KiTTY/releases/download/v$version/klink.exe",
            "https://github.com/cyd01/KiTTY/releases/download/v$version/kscp.exe",
            "https://github.com/cyd01/KiTTY/releases/download/v$version/ksftp.exe"
        ]
    }
```

![image](https://user-images.githubusercontent.com/432127/61544065-f10a6880-aa44-11e9-89b9-fc0d571e5b26.png)

Reply:
Force `$json.autoupdate.PSObject.Properties.Name` be `[Array]` solve the prob, and it could be simplified. Now everything works.